src/modules/SX127x/SX127x.cpp

```diff
@@ -1295,36 +1309,66 @@ RadioLibTime_t SX127x::calculateRxTimeout(RadioLibTime_t timeoutUs) {
   return(numSymbols);
 }
 
-int16_t SX127x::irqRxDoneRxTimeout(uint32_t &irqFlags, uint32_t &irqMask) {
-  // IRQ flags/masks are inverted to what seems logical for SX127x (0 being activated, 1 being deactivated)
-  irqFlags = RADIOLIB_SX127X_MASK_IRQ_FLAG_RX_DEFAULT;
-  irqMask  = RADIOLIB_SX127X_MASK_IRQ_FLAG_RX_DONE & RADIOLIB_SX127X_MASK_IRQ_FLAG_RX_TIMEOUT;
-  return(RADIOLIB_ERR_NONE);
+uint32_t SX127x::getIrqFlags() {
+  return((uint32_t)this->getIRQFlags());
 }
 
-int16_t SX127x::checkIrq(uint8_t irq) {
-  uint16_t flags = getIRQFlags();
-  switch(irq) {
-    case RADIOLIB_IRQ_TX_DONE:
-      return(flags & RADIOLIB_SX127X_CLEAR_IRQ_FLAG_TX_DONE);
-    case RADIOLIB_IRQ_RX_DONE:
-      return(flags & RADIOLIB_SX127X_CLEAR_IRQ_FLAG_RX_DONE);
-    case RADIOLIB_IRQ_HEADER_VALID:
-      return(flags & RADIOLIB_SX127X_CLEAR_IRQ_FLAG_VALID_HEADER);
-    case RADIOLIB_IRQ_CRC_ERR:
-      return(flags & RADIOLIB_SX127X_CLEAR_IRQ_FLAG_PAYLOAD_CRC_ERROR);
-    case RADIOLIB_IRQ_CAD_DONE:
-      return(flags & RADIOLIB_SX127X_CLEAR_IRQ_FLAG_CAD_DONE);
-    case RADIOLIB_IRQ_CAD_DETECTED:
-      return(flags & RADIOLIB_SX127X_CLEAR_IRQ_FLAG_CAD_DETECTED);
-    case RADIOLIB_IRQ_TIMEOUT:
-      return(flags & RADIOLIB_SX127X_CLEAR_IRQ_FLAG_RX_TIMEOUT);
-    default:
-      return(RADIOLIB_ERR_UNSUPPORTED);
+int16_t SX127x::setIrqFlags(uint32_t irq) {
+  // this is a bit convoluted, but unfortunately SX127x IRQ flags are not used to enable/disable that IRQ ...
+  int16_t modem = getActiveModem();
+  if(modem == RADIOLIB_SX127X_LORA) {
+    switch(irq) {
```

Comment:
@jgromes It seems that with this `switch` statement, you cannot set more than one IRQ in the mask, because it checks whether `irq` is exactly equal to one specific flag.

Reply:
That is a good point - unfortunately it also hits a limitation of the SX127x, in which some combinations of IRQ flags cannot be set simultaneously. For example, Tx done and Rx done are both signalled on DIO1 and their configuration is mutually exclusive.

The only (partial) solution I can think of is to do this in a "best effort" approach, and notify the user via a return code (+mabye a debug print) if the combination of IRQ cannot be configured. For example, CAD done and CAD detected 
- which can be a common use case - may be set at the same time because they use a different DIO pin. What do you think?

Reply:
Yes, that sounds reasonable to me.

Reply:
OK, in that case I'll update it accordingly so that we can include it in the 7.0.1 patch ;)

Reply:
@GUVWAF fixed in 38fc7a97a4c195b7c10aa94215a1c53ec18a56ef